active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+> See comments in Markdown for how to use this spec template
+
+<!-- The purpose of this spec is to describe a new feature and
+its APIs that make up a new feature in WinUI. -->
+
+<!-- There are two audiences for the spec. The first are people
+that want to evaluate and give feedback on the API, as part of
+the submission process.  When it's complete
+it will be incorporated into the public documentation at
+docs.microsoft.com (http://docs.microsoft.com/uwp/toolkits/winui/).
+Hopefully we'll be able to copy it mostly verbatim.
+So the second audience is everyone that reads there to learn how
+and why to use this API. -->
+
+# Background
+In WPF Developers had access to Input Validation via [INotifyDataErrorInfo](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifydataerrorinfo?view=netframework-4.7.2) We are adding the same capability to UWP.
```

Comment:
Why not IDataErrorInfo, it's part of .net standard 2.0?

Reply:
INotifyDataErrorInfo is a super set. It's a v2 of IDataErrorInfo, we fell that this covers all of the same scenarios and is async by nature. Are there any scenarios we are not thinking of that are supported in IDataErrorInfo?

Reply:
True, but if I understand correctly, IDataErrorInfo is part of .NET Standard 2.0, and it would be extremely confusing not to support it.

I think you can easily do something like this so it will never cost additional performance:

```
// Async first
var notifyDataErrorInfo = obj as INotifyDataErrorInfo;
if (!(notifyDataErrorInfo is null))
{
    // async validation here
    return;
}

var dataErrorInfo = obj as IDataErrorInfo;
if (!(dataErrorInfo is null))
{
    // sync validation here
    return;
}

// Validation not supported
```

Reply:
Having worked on several complex WPF and Silverlight forms applications, INotifyDataErrorInfo was sufficient for validation purposes.  There was no need for IDataErrorInfo given it was a subset.  There a lot of discussion here around this without making reference to Silverlight.  Silverlight had/has the most evolved validation support given how it nicely tied into UI controls.  Just replicating Silverlight's validation features would be a big step forward for UWP/WPF/Xamarin.